generator/src/main/java/com/scottlogic/deg/generator/restrictions/StringRestrictionsFactory.java

```diff
@@ -0,0 +1,76 @@
+package com.scottlogic.deg.generator.restrictions;
+
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+public class StringRestrictionsFactory {
+    public StringRestrictions forStringMatching(Pattern pattern, boolean negate) {
```

Comment:
Some ugly constructor use in here, ~object initialiser~ builder?

Reply:
Not sure I follow this comment, what is your concern?

Reply:
It's very difficult to tell what "new TextualRestrictions(null,null,Colelctions.EmptySet()...." is supposed to mean for any of these following methods. It would be better if the parameters were named when passed to TextualRestrictions for readability sake. The only way we can do this in java is use a builder for TextualRestrictions?

Reply:
@MattCline-SL - An alternative approach is to include these methods as factory-methods on `TextualRestrictions` and make the constructor private.

This - funnily enough - is what I originally had before some previous PR comments.

It would then look like `TextualRestrictions.forStringMatching(...)`